File: Searching Algorithms/Binary Search/binary-search.java
import java.util.Scanner;
public class binary_search{
    public static void main(String[] args){
        int first,last,mid=0,n,i,element,flag=0;
        Scanner s=new Scanner(System.in);
        System.out.print('\u000C');
        System.out.println("Enter the number of elements you are going to enter:");
        n=s.nextInt();
        int arr[]=new int[n];
        System.out.println("Enter the elements in sorted order:");
        for(i=0;i<n;i++){
            arr[i]=s.nextInt();
        }
        System.out.println("Enter the element that you wish to search:");
        element=s.nextInt();
        first=0;
        last=n-1;
        while(first<=last){
            mid=(first+last)/2;
            if(arr[mid]==element){
                flag=1;
                break;
            }
            else if(element>arr[mid]){
                first=mid+1;
            }
            else{
                last=mid-1;
            }
        }
        if(flag==1){
            System.out.println("Element is present at "+(mid+1)+" position");
        }
        else{
            System.out.println("Element is not present");
        }
    }
}
        

File: Searching Algorithms/Binary Search/binary-search.c
#include <stdio.h>
#include <stdlib.h>
int main(){
  int arr[1000],n,i,element;
  int first,last,mid,flag=0;
  printf("Enter the number of elements in the array: \n");
  scanf("%d",&n);
  printf("Enter the elements in the array: \n");
  for(i=0;i<n;i++){
    scanf("%d",&arr[i]);
  }
  printf("Enter the element that has to be searched: \n");
  scanf("%d",&element);
  first=0;
  last=n-1;
  while(first<=last){
    mid=(first+last)/2;
      
    if(arr[mid]<element){
      first=mid+1;
    }
    else if(arr[mid]==element){
      flag=1;
      break;
    }
    else{
      last=mid-1;
    }
    
  }
  if(flag==1){
    printf("Element has been found at position %d \n",mid+1);
  }
  else{
      printf("Element has not been found");
  }
  return 0;

}
